### PR TITLE
Feat/notifications

### LINKED ISSUE #60: Notifications and layot enhancements.
Add notifications (backend, frontend).
Change font to IBM Plex Mono.
Enhance menu layout.

### PR DESCRIPTION
**Pull request by aleksikosonen**
fixes #60

## Description

Created front and backend for notifications when a new device announces itself. Worked on this together with @jaanikaukonen 

Requires `npm i` in backend folder

### Frontend
- Created `NotificationCenterComponent` and `NotificationComponent`
- Added `notificationUrl` to `globals.ts`
- Added `useNotifications` to `ApiHooks.ts`
- Added `notificationType`
- Implemented logic to `BaseView` to open the NotificationCenter

### Backend
- Added `deviceNotificationController`, `deviceNotificationRoutes`, schema and for `deviceNotification`
- Create new Notification in `mqttServices.ts`
- Added `cors` library to handle endpoint headers on, used on `app.ts`

### Additional
- Fixed bug in DetailView by passing device.sensors as string
- Added `Google IBM Plex Mono` font
- Deleted console.log that printed the word 'error' in `mqtt.services`

### CODE REVIEW

**Review by aleksikyto — APPROVED** [review]
LGTM

File: back-end/src/controllers/deviceNotificationController.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+export const deleteDeviceNotification = async (req, res) => {
+    try {
+        const notification = await DeviceNotification.findByIdAndDelete(req.params.id)
+        res.status(200).json(notification)
+    } catch (error) {
+        console.log(error)
+        res.status(400)
+    }
+}
```
Comment (NikoLindborg) [inline back-end/src/controllers/deviceNotificationController.ts:54]: All the exports should ideally be defined in the last line of the file. We should change that for the whole entire app.

File: back-end/src/mqtt/mqttServices.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import { DeviceNotification } from './../schemas/DeviceNotification';
```
Comment (NikoLindborg) [inline back-end/src/mqtt/mqttServices.ts:1]: Unify spaces with the rest of the imports

File: back-end/src/routes/deviceNotificationRoutes.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { deleteAllNotifications, deleteDeviceNotification, getDeviceNotifications, setDeviceNotification } from './../controllers/deviceNotificationController';
+import express from 'express'
+
+
+export const router = express.Router()
+
+router.route('/').get(getDeviceNotifications).post(setDeviceNotification).delete(deleteAllNotifications)
+router.route('/:id').delete(deleteDeviceNotification)
```
Comment (NikoLindborg) [inline back-end/src/routes/deviceNotificationRoutes.ts:8]: All the different http requests could be started on a new line
router.route('/')  
 .get(getDeviceNotifications)
         .post(setDeviceNotification)
    .delete(deleteAllNotifications)

**Review by NikoLindborg — CHANGES_REQUESTED** [review]
Just do the few small changes and should be good to go

File: back-end/src/app.ts [hunk 4]
```diff
@@ -5,16 +5,20 @@ const port = 3001
...
 import {router as channelRoutes} from './routes/channelRoutes'
+import {router as deviceNotificationRoutes} from './routes/deviceNotificationRoutes'
 import WebSocketClient from './websocket/websocket'
 import mqttClient from './mqtt/mqtt'
 
 connectDB()
 
 console.log('The WebSocket server is running on port 8080')
 const mqtt = mqttClient()
+var cors = require('cors')
```
[the remark this answers is not in the document]
Comment (aleksikosonen) [reply back-end/src/app.ts:16]: Fixed

File: back-end/src/controllers/deviceNotificationController.ts (hunk 1 above)
Comment (aleksikosonen) [reply back-end/src/controllers/deviceNotificationController.ts:54]: That should be done in seperate PR. Not related to notifications, please submit new issue.

File: back-end/src/mqtt/mqttServices.ts (hunk 2 above)
Comment (aleksikosonen) [reply back-end/src/mqtt/mqttServices.ts:1]: Fixed

File: back-end/src/routes/deviceNotificationRoutes.ts (hunk 3 above)
Comment (aleksikosonen) [reply back-end/src/routes/deviceNotificationRoutes.ts:8]: Fixed

**Review by NikoLindborg — APPROVED** [review]
looks good now

**Pull request merged**